### ISSUE TITLE
Building Flow 0.21.0 on NixOS requires OCaml 4.02, not OCaml 4.01

### ISSUE THREAD

**Issue by RocketPuppy** [facebook/flow#1398]
The current flow documentation lists OCaml 4.01 as the required OCaml version, however I was unable to build the current release of Flow 0.21.0 without OCaml 4.02. I'm unsure if this is specifically related to NixOS or if the build instructions actually need updated.

**Comment by jeffmo** [comment]
Hmm... We're pretty careful to ensure that Flow still builds on 4.01 by way of [Travis CI](https://travis-ci.org/facebook/flow/builds) (because that's the only version available in some LTS distros).

Can you walk me through the steps you tried that didn't work?

**Comment by RocketPuppy** [comment]
First thing I did was check out the code from Github and checked out the v0.21.0 branch. After that I made sure I had libelf and ocaml set up as those were the dependencies specified in the README. Running `ocamlbuild -version` gave me `ocamlbuild 4.01.0`, which seems to check out as far as dependencies go.

Finally I ran make. It warned about not having `hg` and `tput` but proceeded to build anyway until it failed with the following error:
```
File "src/common/files_js.ml", line 78, characters 6-15:
Error: Syntax error: operator expected.
```

Next I updated ocaml to `ocamlbuild 4.02.3`. Then ran `make clean`, followed by `make`. The build still warned about not having `hg` and `tput`, but finished successfully putting the `flow` binary in `./bin`.

Let me know if you need any more information.

**Comment by jeffmo** [comment]
Aha! Yes, I think this was an issue specific to when we cut the 0.21 branch (we were using some syntax that wasn't supported by 4.01, only 4.02). See https://github.com/facebook/flow/commit/d2bd1fee3aa3c2cfdd60b8f1e0094969d1815308 which fixes this (but landed *just* after 0.21 was cut).

Sorry about the confusion here -- we'll try to make sure the build is working for all versions when we cut in the future.

**Comment by RocketPuppy** [comment]
No problem, thanks for your time.
